### PR TITLE
refactor `rio_tiler.utils.tile_exits` to allow raster bounds latitude == -90,90

### PR DESCRIPTION
**Pull request by vincentsarago**
This pr refactor `rio_tiler.utils.tile_exits` to make sure rio-tiler doesn't fails when a raster as a bounds == -90 or 90 latitude 

cc @abarciauskas-bgse

### CODE REVIEW

**Comment by kylebarron** [comment]
Is this `contains` or `intersects`?

**Comment by vincentsarago** [comment]
@kylebarron good question

**Comment by kylebarron** [comment]
From a cursory look it looked like contains

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,7 @@
+2.0a11 (2020-05-29)
+------------------
+- refactor `rio_tiler.utils.tile_exits` to allow raster bounds latitude == -90,90
```
Comment (abarciauskas-bgse) [inline CHANGES.txt:3]: Minor but I think this should be `tile_exists`

Comment (kylebarron) [reply CHANGES.txt:3]: I was pretty confused about where the tiles were going!

**Comment by vincentsarago** [comment]
thanks @kylebarron and @abarciauskas-bgse 
I added more tests to confirm the code does intersects and contains 🎉

**Pull request merged**